From 91e0db96741359173ddf2be083aafcc1a3c32472 Mon Sep 17 00:00:00 2001
From: Jiaqing Zhao <jiaqing.zhao@linux.intel.com>
Date: Mon, 18 Apr 2022 11:22:43 +0800
Subject: [PATCH] Disable shell for default users

Change the shell of all global static users other than root (which
retains /bin/sh) and sync (as /bin/sync is rather harmless) to
/sbin/nologin (as /usr/sbin/nologin does not exist in openembedded)

Upstream-Status: Backport [https://launchpad.net/ubuntu/+source/base-passwd/3.5.30]
Signed-off-by: Jiaqing Zhao <jiaqing.zhao@linux.intel.com>
---
 passwd.master | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/passwd.master b/passwd.master
index e1c32ff..0cd5ffd 100644
--- a/passwd.master
+++ b/passwd.master
@@ -1,18 +1,18 @@
 root::0:0:root:/root:/bin/sh
-daemon:*:1:1:daemon:/usr/sbin:/bin/sh
-bin:*:2:2:bin:/bin:/bin/sh
-sys:*:3:3:sys:/dev:/bin/sh
+daemon:*:1:1:daemon:/usr/sbin:/sbin/nologin
+bin:*:2:2:bin:/bin:/sbin/nologin
+sys:*:3:3:sys:/dev:/sbin/nologin
 sync:*:4:65534:sync:/bin:/bin/sync
-games:*:5:60:games:/usr/games:/bin/sh
-man:*:6:12:man:/var/cache/man:/bin/sh
-lp:*:7:7:lp:/var/spool/lpd:/bin/sh
-mail:*:8:8:mail:/var/mail:/bin/sh
-news:*:9:9:news:/var/spool/news:/bin/sh
-uucp:*:10:10:uucp:/var/spool/uucp:/bin/sh
-proxy:*:13:13:proxy:/bin:/bin/sh
-www-data:*:33:33:www-data:/var/www:/bin/sh
-backup:*:34:34:backup:/var/backups:/bin/sh
-list:*:38:38:Mailing List Manager:/var/list:/bin/sh
-irc:*:39:39:ircd:/var/run/ircd:/bin/sh
-gnats:*:41:41:Gnats Bug-Reporting System (admin):/var/lib/gnats:/bin/sh
-nobody:*:65534:65534:nobody:/nonexistent:/bin/sh
+games:*:5:60:games:/usr/games:/sbin/nologin
+man:*:6:12:man:/var/cache/man:/sbin/nologin
+lp:*:7:7:lp:/var/spool/lpd:/sbin/nologin
+mail:*:8:8:mail:/var/mail:/sbin/nologin
+news:*:9:9:news:/var/spool/news:/sbin/nologin
+uucp:*:10:10:uucp:/var/spool/uucp:/sbin/nologin
+proxy:*:13:13:proxy:/bin:/sbin/nologin
+www-data:*:33:33:www-data:/var/www:/sbin/nologin
+backup:*:34:34:backup:/var/backups:/sbin/nologin
+list:*:38:38:Mailing List Manager:/var/list:/sbin/nologin
+irc:*:39:39:ircd:/var/run/ircd:/sbin/nologin
+gnats:*:41:41:Gnats Bug-Reporting System (admin):/var/lib/gnats:/sbin/nologin
+nobody:*:65534:65534:nobody:/nonexistent:/sbin/nologin
-- 
2.32.0

